static/app/views/dashboards/dashboardLegendUtils.tsx

```diff
@@ -0,0 +1,204 @@
+import type {Location} from 'history';
+
+import type {Series} from 'sentry/types/echarts';
+import type {InjectedRouter} from 'sentry/types/legacyReactRouter';
+import type {Organization} from 'sentry/types/organization';
+import {decodeList} from 'sentry/utils/queryString';
+
+import {type DashboardDetails, DisplayType, type Widget} from './types';
+
+type Props = {
+  dashboard: DashboardDetails | null;
+  location: Location;
+  organization: Organization;
+  router: InjectedRouter;
+};
+
+class DashboardLegendEncoderDecoder {
```

Comment:
Accurate name, but I think `WidgetLegendState` or `WidgetLegendSelectionState` would be more evocative! `...State` implies that the class is used to manage a state, which is true! Plus, removing "Dashboard" from the name is good, since we'll probably want to use this for non-Dashboard UIs in the future!

Oh also, the file name should match, so `widgetLegendState.tsx` or whichever you go with